libcxx/include/stdalign.h

```diff
@@ -0,0 +1,42 @@
+// -*- C++ -*-
```

Comment:
We don't need a wrapper here, since clang provides a correct one.

Reply:
Seems needed to make CI pass.

Reply:
Could you explain?

Reply:
When I attempted to just remove `stdaligh.h`, there was build error. Perhaps CMakelist.txt should also be modified.

I'm not sure whether we can/should just be inconsistent for `stdaligh.h`, given libc++ is providing almost all wrapper C headers required by the C++ standard library.

Reply:
Given that we didn't have `stdalign.h` before there shouldn't be a build error. Also note that we only provide C headers which we have to for conformance. We don't provide `assert.h` or `limits.h` for example.

Reply:
We certainly do provide stdatomic.h

Reply:
@jwakely Sorry, I meant `stdalign.h`. At least it looks to me like you're not providing the required macros when including it.

Reply:
Yeah, GCC's `stdalign.h` only defines them for C11 and C17, but no versions of C++. I have a local patch fixing it which I'll propose. I don't think anybody has ever noticed before, because nobody cares about those macros in C++ (so much so that one of them was overlooked until earlier this year and nobody noticed or cared). The `__alignof_is_defined` macro isn't actually mentioned in any published C++ standard.

Reply:
Patch to fix GCC's `<stdalign.h>`: https://gcc.gnu.org/pipermail/gcc-patches/2024-October/666236.html

Thanks for pointing it out!